components/brave_wallet_ui/components/desktop/wallet-menus/wallet_settings_menu.tsx

```diff
@@ -85,8 +85,20 @@ export const WalletSettingsMenu = (props: Props) => {
       return
     }
 
-    const route =
-      selectedNetwork.coin === BraveWallet.CoinType.ETH ? 'ethereum' : 'solana'
+    let route: string
+    switch (selectedNetwork.coin) {
+      case BraveWallet.CoinType.ETH:
+        route = 'ethereum'
+        break
+      case BraveWallet.CoinType.SOL:
+        route = 'solana'
+        break
+      case BraveWallet.CoinType.ADA:
+        route = 'cardano'
+        break
+      default:
+        throw new Error('Coin not supported')
```

Comment:
This is typically asserted as 
```
assertNotReached(`Unknown coin ${coin}`)
```
Not sure if `throw` is enough
cc: @onyb @Douglashdaniel